azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2UserService.java

```diff
@@ -74,8 +74,9 @@ public OidcUser loadUser(OidcUserRequest userRequest) throws OAuth2Authenticatio
 
         // Create a copy of oidcUser but use the mappedAuthorities instead
         oidcUser = new DefaultOidcUser(mappedAuthorities, oidcUser.getIdToken(), getUserNameAttrName(userRequest));
+        final GraphOidcUser graphOidcUser = new GraphOidcUser(oidcUser, graphApiToken);
```

Comment:
In this area, there is the same problem with the problem in AADAuthenticationFilter.java this Class, you shouldn't put graphToken in the OidcUser object, if you need to use it, you can get a new one again with the com.microsoft.azure.spring.autoconfigure.aad.AzureADGraphClient#acquireTokenForGraphApi method. What do you think about it?

Reply:
similarly as above, in order to be able to get the token from the principal: 
GraphOidcUser graphUser = (GraphOidcUser)SecurityContextHolder.getContext().getAuthentication().getPrincipal();

String token = graphUser.getGraphApiToken();

please let me know your thoughts/if this should be removed (similarly for the above one)

Reply:
Change the OidcUser's implementation from DefaultOidcUser to GraphOidcUser  is not safe, it may cause some incompatibility problem for the old user who uses like this:
DefaultOidcUser graphUser = (DefaultOidcUser)SecurityContextHolder.getContext().getAuthentication().getPrincipal();
And also the token should not be put in the OidcUser, it's not safe.

If you want to get the graphAPIs token. You have many choices to get it:
method one:
 you can put graphAPIs token in the session, than you can get it from the session just like the way we do in AADAuthenticationFilter.class

method two:
you can get the graphAPIs token by AzureADGraphClient#acquireTokenForGraphApi:
graphApiToken = graphClient.acquireTokenForGraphApi(idToken.getTokenValue(),aadAuthProps.getTenantId()).getAccessToken();

Reply:
ok makes sense, have gone ahead and removed GraphOidcUser